scripts/config-plugins/plugins/cocoaBaseMods.mjs

```diff
@@ -0,0 +1,53 @@
+// @ts-check
+import { createRequire } from "node:module";
+import * as path from "node:path";
+import { BaseMods } from "../ExpoConfigPlugins.mjs";
+import { makeFilePathModifier, makeNullProvider } from "../provider.mjs";
+
+const require = createRequire(import.meta.url);
+
+/**
+ * @param {import("../types.js").CustomModProvider} modifyFilePath
+ * @returns {import("../types.js").IosModFileProviders}
+ */
+export function createModFileProviders(modifyFilePath) {
+  const modifyReactNativeHostFilePath = makeFilePathModifier(
+    path.dirname(require.resolve("@rnx-kit/react-native-host/package.json"))
+  );
+
+  const nullProvider = makeNullProvider();
+
+  // https://github.com/expo/expo/blob/93cd0503117d5a25f8b80ed7b30ec5bed3a67c24/packages/@expo/config-plugins/src/plugins/withIosBaseMods.ts
+  const expoProviders = BaseMods.getIosModFileProviders();
+
+  /** @type {import("../types.js").IosModFileProviders} */
+  const defaultProviders = {
+    dangerous: expoProviders.dangerous,
+    finalized: expoProviders.finalized,
+    appDelegate: modifyFilePath(
+      expoProviders.appDelegate,
+      "ReactTestApp/AppDelegate.swift"
+    ),
+    expoPlist: nullProvider,
+    xcodeproj: modifyFilePath(
+      expoProviders.xcodeproj,
+      "ReactTestApp.xcodeproj/project.pbxproj"
+    ),
+    infoPlist: modifyFilePath(expoProviders.infoPlist, "Info.plist"),
+    entitlements: nullProvider,
+    podfile: makeNullProvider({
+      path: "",
+      language: /** @type {const} */ ("rb"),
+      contents: "",
+    }),
+    podfileProperties: makeNullProvider(),
+  };
```

Comment:
This is really clever. 👏 

Given RNTA has a predictable structure, a lot of the complexity in the plugins (e.g. searching for all possible locations for an [Info.plist](https://github.com/expo/expo/blob/ea5d3f54cbc93cb7d84f9bb53af62d5f6cebfe84/packages/%40expo/config-plugins/src/ios/Paths.ts#L256) or [Xcodeproj](https://github.com/expo/expo/blob/ea5d3f54cbc93cb7d84f9bb53af62d5f6cebfe84/packages/%40expo/config-plugins/src/ios/Paths.ts#L180)) can be totally sidestepped.

I am not sure to what extent this sidesteps all of their hard-coding of the `ios` platform, but given how your test passes, maybe it's quite a lot. Surely at least one of these will end up referring to `config.ios` or the `ios` folder at some point, though? Do you have an impression of the extent to which Expo config plugins will work?

Reply:
As far as I've understood, the providers are just functions for making modifications. They don't have props that are platform specific. The `with*` functions (e.g. `withAppDelegate`) are platform specific. For those, we need a separate set for macOS, and that's what I've done here: https://github.com/microsoft/react-native-test-app/blob/90ae7901ce523404faa31a62bf0b58a133d4125f/plugins/index.js#L15-L55

Which means that for now, we only support these four functions. And you will need to import them from `react-native-test-app/plugins/index.js`.